Add RELEASE_LOG_FAULT in Connection::markCurrentlyDispatchedMessageAsInvalid

achristensen07 opened the pull request:
#### 3c14d9145daf10453202e4f2e0e6259f325ba9b3
<pre>
Add RELEASE_LOG_FAULT in Connection::markCurrentlyDispatchedMessageAsInvalid
<a href="https://bugs.webkit.org/show_bug.cgi?id=247138">https://bugs.webkit.org/show_bug.cgi?id=247138</a>
rdar://101641127

Reviewed by David Kilzer.

* Source/WebKit/Platform/IPC/Connection.cpp:
(IPC::Connection::markCurrentlyDispatchedMessageAsInvalid):

Canonical link: <a href="https://commits.webkit.org/256087@main">https://commits.webkit.org/256087@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5a9e8054bd6c627bde0c5ebc52daf1e069465da7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/94602 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/3764 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/27485 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/104229 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/3828 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/31955 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/86908 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/100197 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/100269 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/2753 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/80971 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/29770 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/84678 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/84235 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/72665 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/38359 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/18056 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/36215 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/19330 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/40118 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/41979 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/1998 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/42084 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/38559 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

ddkilzer commented on Source/WebKit/Platform/IPC/Connection.cpp:
```diff
@@ -501,6 +501,8 @@ void Connection::invalidate()
 
 void Connection::markCurrentlyDispatchedMessageAsInvalid()
 {
+    RELEASE_LOG_FAULT(IPC, "Invalid message dispatched");
```
Nit:  "Invalid IPC message dispatched".

The other option here would be to put in the `MESSAGE_CHECK_*` macros in `Source/WebKit/Platform/IPC/Connection.h` and do this to get unique signatures:

```
    RELEASE_LOG_FAULT(IPC, WTF_PRETTY_FUNCTION ": Invalid message dispatched");
```

I would actually prefer that because we'd get unique signatures for each issue.

ddkilzer replied:
As the patch currently stands, all issues from all message checks would be aggregated under a single "Invalid message dispatched" signature.

ddkilzer reviewed (approved):
r=me, but consider the alternative proposed.

Commit:
Add RELEASE_LOG_FAULT in Connection::markCurrentlyDispatchedMessageAsInvalid
https://bugs.webkit.org/show_bug.cgi?id=247138
rdar://101641127

Reviewed by David Kilzer.

* Source/WebKit/Platform/IPC/Connection.cpp:
(IPC::Connection::markCurrentlyDispatchedMessageAsInvalid):

Canonical link: https://commits.webkit.org/256087@main
